.github/workflows/build-test.yml

```diff
@@ -9,29 +9,13 @@ jobs:
   prepare:
     name: Prepare
     runs-on: ubuntu-latest
-    outputs:
-      YARN_CACHE_DIR: ${{ steps.yarn-cache-dir.outputs.YARN_CACHE_DIR }}
-      YARN_VERSION: ${{ steps.yarn-version.outputs.YARN_VERSION }}
-    strategy:
-      matrix:
-        node-version: [14.x, 16.x, 18.x, 19.x]
```

Comment:
What's behind this change? node caches are in general not compatible between major versions.

https://github.com/actions/setup-node#matrix-testing

Reply:
(In either case, if the intention is to build a single version only, I guess that's significant enough to belong in a separate PR?)

Reply:
We're caching the Yarn cache directory here, which is a collection of .zip files that represent exactly what was downloaded from the package repository. I thought this cache would work the same across Node.js versions. I'm not certain though.

The only Node.js-version-specific steps I'm aware of are postinstall steps (e.g. especially for native addons that use the Node.js C/C++ API). So caching `node_modules` would definitely break in some cases between Node.js versions.

Reply:
I disagree with this change as well. I think we should install Yarn dependencies for each Node version rather than for the development version in `.nvmrc` once. It may be that some of the dependencies in `package.json` are only compatible with the development version or higher, but any incompatibilities with versions less than the development version won't be caught in CI if we simply reuse the development Node version when installing dependencies.

Reply:
This should be caught by the other jobs, as they still run `yarn install` for each specific Node version.

Reply:
Alright, sounds good.